akka-actor/src/main/scala/akka/dispatch/affinity/AffinityPool.scala

```diff
@@ -68,42 +68,40 @@ private[affinity] object AffinityPool {
     private[this] val maxParkPeriodNs = MICROSECONDS.toNanos(280 - 30 * idleCpuLevel)
 
     private[this] var state: IdleState = Initial
-    private[this] var spins = 0L
-    private[this] var yields = 0L
+    private[this] var turns = 0L
     private[this] var parkPeriodNs = 0L
 
+    @inline private[this] final def transitionTo(newState: IdleState): Unit = {
+      state = newState
+      turns = 0
+    }
+
     def idle(): Unit = {
       (state: @switch) match {
```

Comment:
@viktorklang. What if `turns` is renamed to `turnsInState` and the increment operation gets moved right above the switch statement. Would feel a lot more "FSM like" to me. Right now it looks a bit confusing that `turns` tracks the number of consecutive state steps but only when we are in `Spinning` and `Yielding`. I think this will make it easier to grasp.  Despite that, the actual value could be used in some statistics at a certain point to show things like consecutive parks.